vars/UpdateBuildFailureIssues.groovy

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+ /** Library to create, update an close GitHub issue across opensearch-project repositories for distribution build failures.
+ @param Map args = [:] args A map of the following parameters
+ @param args.failureMessages <required> - Failure message retrieved from buildFailureMessage() method.
+ @param args.passMessages <required> - Passing message retrieved from buildFailureMessage() method.
+ @param args.inputManifestPath <required> - Path to input manifest.
+ */
+void call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+    def failureMessages = args.failureMessages
+    def passMessages = args.passMessages
+    def yamlFile = readYaml(file: args.inputManifestPath)
+    def currentVersion = yamlFile.build.version
+
+    List<String> failedComponents = []
+    List<String> passedComponents = []
+
+    for (message in failureMessages) {
+        java.util.regex.Matcher match = (message =~ /(?<=\bError building\s).*/)
+        String matched = match[0]
+        failedComponents.add(matched.split(' ')[0].split(',')[0].trim())
+    }
+    failedComponents = failedComponents.unique()
+    for (message in passMessages) {
+        java.util.regex.Matcher match = (message =~ /(?<=\bSuccessfully built\s).*/)
+        String matched = match[0]
+        passedComponents.add(matched.split(' ')[0].trim())
+    }
+    passedComponents = passedComponents.unique()
+
+    for (component in yamlFile.components) {
+        if (failedComponents.contains(component.name)) {
+            println("Component ${component.name} failed, creating github issue")
+            exactComponentFailureMessage = getExactErrorMessage(failureMessages, component.name)
+            ghIssueBody = """***Received Error***: **${exactComponentFailureMessage}**.
+                    ${component.name} failed during the distribution build for version: ${currentVersion}.
+                    Please see build log at ${env.RUN_DISPLAY_URL}.
+                    The failed build stage will be marked as unstable(!). Please see ./build.sh step for more details""".stripIndent()
+            createGithubIssue(
+                repoUrl: component.repository,
+                issueTitle: "[AUTOCUT] Distribution Build Failed for ${component.name}-${currentVersion}",
+                issueBody: ghIssueBody,
+                label: "autocut,v${currentVersion}"
+                )
+            sleep(time:3, unit:'SECONDS')
+        } else if (passedComponents.contains(component.name) && !failedComponents.contains(component.name)) {
```

Comment:
What is the case for `else` here?

Reply:
Nothing. The component will either pass or fail.

Reply:
So then the logic has issues.

If there is anything in failed, it will always go to the `if` block, and ignore `else if` block.

Therefore, you can change the `else if` block to just `else` block?

Reply:
No! If the component fail by default it will go in if block and if it passes completely, it will go in else if block. I can add an else block just to print something but the component will either pass or fail. There is no other possible outcome.

Reply:
After the talk with @gaiksaya offline, this can be improve to be two `if` blocks:

`1st         if (failedComponents.contains(component.name)) {`
will check if the component has a failure. If so, create issue or keep updating comment.

`2nd if (passedComponents.contains(component.name) && !failedComponents.contains(component.name)) {`
will check if component shows up in passes but clear of failed, close issues.

For any other cases such as pass is empty or in pass but also in fail, ignore.

Thanks.